internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -0,0 +1,671 @@
+		Name:  assistedServiceContainerName,
+		Image: ServiceImage(),
+		Ports: []corev1.ContainerPort{
+			{
+				ContainerPort: servicePort,
+				Protocol:      corev1.ProtocolTCP,
+			},
+		},
+		Env: serviceEnv,
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "bucket-filesystem",
+				MountPath: "/data",
+			},
+		},
+		Resources: corev1.ResourceRequirements{
+			Limits: corev1.ResourceList{
+				corev1.ResourceCPU:    *resource.NewMilliQuantity(500, resource.DecimalSI),
+				corev1.ResourceMemory: *resource.NewQuantity(2000*1024*1024, resource.BinarySI),
+			},
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    *resource.NewMilliQuantity(300, resource.DecimalSI),
+				corev1.ResourceMemory: *resource.NewQuantity(400*1024*1024, resource.BinarySI),
+			},
+		},
+		LivenessProbe: &corev1.Probe{
+			FailureThreshold:    3,
+			SuccessThreshold:    1,
+			InitialDelaySeconds: 3,
+			PeriodSeconds:       10,
+			TimeoutSeconds:      3,
+			Handler: corev1.Handler{
+				TCPSocket: &corev1.TCPSocketAction{
+					Port: intstr.FromInt(int(servicePort)),
+				},
+			},
+		},
+	}
+
+	postgresContainer := corev1.Container{
+		Name:            databaseContainerName,
+		Image:           DatabaseImage(),
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Ports: []corev1.ContainerPort{
+			{
+				Name:          databaseName,
+				ContainerPort: databasePort,
+				Protocol:      corev1.ProtocolTCP,
+			},
+		},
+		Env: []corev1.EnvVar{
+			newSecretEnvVar("POSTGRESQL_DATABASE", "db.name", databaseSecretName),
+			newSecretEnvVar("POSTGRESQL_USER", "db.user", databaseSecretName),
+			newSecretEnvVar("POSTGRESQL_PASSWORD", "db.password", databaseSecretName),
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "postgresdb",
+				MountPath: "/var/lib/pgsql/data",
+			},
+		},
+		Resources: corev1.ResourceRequirements{
+			Limits: corev1.ResourceList{
+				corev1.ResourceCPU:    *resource.NewMilliQuantity(200, resource.DecimalSI),
+				corev1.ResourceMemory: *resource.NewQuantity(500*1024*1024, resource.BinarySI),
+			},
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    *resource.NewMilliQuantity(100, resource.DecimalSI),
+				corev1.ResourceMemory: *resource.NewQuantity(400*1024*1024, resource.BinarySI),
+			},
+		},
+	}
+
+	volumes := []corev1.Volume{
+		{
+			Name: "bucket-filesystem",
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: filesystemPVCName,
+				},
+			},
+		},
+		{
+			Name: "postgresdb",
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: databasePVCName,
+				},
+			},
+		},
+	}
+
+	deploymentLabels := map[string]string{
+		"app": assistedServiceDeploymentName,
+	}
+
+	deploymentStrategy := appsv1.DeploymentStrategy{
+		Type: appsv1.RollingUpdateDeploymentStrategyType,
+		RollingUpdate: &appsv1.RollingUpdateDeployment{
```

Comment:
We want to have either 0 or 1 running at a time, right? Should we just go with the recreate strategy?